auparse/auparse.go

```diff
@@ -66,10 +65,31 @@ type field struct {
 	value string // Parsed and enriched value.
 }
 
-func newField(orig string) *field { return &field{orig: orig, value: orig} }
-func (f *field) Orig() string     { return f.orig }
-func (f *field) Value() string    { return f.value }
-func (f *field) Set(value string) { f.value = value }
+func newField(orig string) field { return field{orig: orig, value: orig} }
+
+type fieldMap map[string]field
+
+func (fm fieldMap) add(key string, f field) {
+	fm[key] = f
+}
+
+func (fm fieldMap) setFieldValue(key, value string) {
+	if f, ok := fm[key]; ok {
+		fm[key] = field{f.orig, value}
```

Comment:
This *is* a local type, but the fields are the same type, so please add explicit field names.